src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -97,6 +98,10 @@ internal PackageSourceMoniker SelectedSource
 
         private readonly Guid _sessionGuid = Guid.NewGuid();
         private readonly Stopwatch _sinceLastRefresh;
+        private readonly Stopwatch _sinceUserAction;
+        private PerformanceMetrics _performanceMetrics;
+        private bool _installedTabIsLoaded;
```

Comment:
Flags I'm using to indicate whether installed and updates tabs have been preloaded. If so, from now on they are eligible for UI filtering.